### PR TITLE
WM-1606 - Add nonterminal state

### PR DESCRIPTION
**Pull request by kpierre13**

### CODE REVIEW

File: src/pages/SubmissionDetails.js [hunk 1]
```diff
@@ -193,15 +219,17 @@ export const SubmissionDetails = ({ submissionId }) => {
                 field: 'state',
                 headerRenderer: () => h(Sortable, { sort, field: 'state', onSort: setSort }, ['Status']),
                 cellRenderer: ({ rowIndex }) => {
+                  const getStatus = state(paginatedPreviousRuns[rowIndex].state)
```
Comment (mspector) [inline src/pages/SubmissionDetails.js:222]: nitpicking / thinking out loud - maybe this variable should be called `status`? if you agree, I would suggest it. not a big deal, though.

**Review by mspector — COMMENTED** [review]
code looks good to me -- would you mind posting a screenshot or two?

**Review by mspector — APPROVED** [review]
LGTM. 

Not asking for a change -- I'm interested in the opinions of others here -- but I kind of like the clock icon for "initializing" instead of the gear icon. Other reviewers and @kpierre13 , let me know what you think...

**Comment by kpierre13** [comment]
> LGTM.
> 
> Not asking for a change -- I'm interested in the opinions of others here -- but I kind of like the clock icon for "initializing" instead of the gear icon. Other reviewers and @kpierre13 , let me know what you think...

I'm good with either. That is what I had originally, but purely chose the cog for sake of variety. I'm happy to go with what majority thinks.

File: src/components/job-common.js [hunk 2]
```diff
@@ -23,19 +23,44 @@ export const statusType = {
...
   },
+  queued: {
+    id: 'queued', // Must match variable name for collection unpacking.
+    label: () => 'Queued',
+    icon: style => icon('clock', { size: iconSize, style: { color: colors.dark(), ...style } })
+  },
+  initializing: {
+    id: 'initializing', // Must match variable name for collection unpacking.
+    label: () => 'Initializing',
+    icon: style => icon('cog', { size: iconSize, style: { color: colors.dark(), ...style } })
```
Comment (cjllanwarne) [inline src/components/job-common.js:57]: To me initializing is a special form of "Running" so I would match that icon here

File: src/components/job-common.js [hunk 3]
```diff
@@ -23,19 +23,44 @@ export const statusType = {
...
+  },
+  canceling: {
+    id: 'canceling', // Must match variable name for collection unpacking.
+    label: () => 'Canceling',
+    icon: style => icon('cog', { size: iconSize, style: { color: colors.dark(), ...style } })
```
Comment (cjllanwarne) [inline src/components/job-common.js:62]: Could probably also match running, or ask for UX help. I would avoid adding a new icon without consultation but re-using an existing one should be fine.

File: src/pages/SubmissionDetails.js [hunk 4]
```diff
@@ -59,6 +59,32 @@ export const SubmissionDetails = ({ submissionId }) => {
     return filterStatement
   }
 
+  //   CANCELING("CANCELING");
```
Comment (cjllanwarne) [inline src/pages/SubmissionDetails.js:62]: This comment looks out of place now?

File: src/pages/SubmissionDetails.js [hunk 5]
```diff
@@ -59,6 +59,32 @@ export const SubmissionDetails = ({ submissionId }) => {
...
+
+  const state = state => {
+    switch (state) {
+      case 'SYSTEM_ERROR':
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply src/pages/SubmissionDetails.js:66]: In general I think all cases fall through to the next one unless they either exit (eg with a `return`) or explicitly `break`... but that might just be java/c

**Review by cjllanwarne — APPROVED** [review]
LGTM, pending a few icon updates

**Pull request merged**